### PR TITLE
New Trim functions

### PR DESCRIPTION
**Pull request by gem-neo4j**

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-936.surge.sh

Updated pages:
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-936.surge.sh/cypher-manual/5/deprecations-additions-removals-compatibility)
- [Functions](https://neo4j-docs-cypher-936.surge.sh/cypher-manual/5/functions)
- [String functions](https://neo4j-docs-cypher-936.surge.sh/cypher-manual/5/functions/string)

**Review by loveleif — APPROVED** [review]

File: modules/ROOT/pages/functions/string.adoc [hunk 1]
```diff
@@ -18,6 +18,74 @@ This `STRING` will therefore be formatted according to the https://en.wikipedia.
 
 See also xref::syntax/operators.adoc#query-operators-string[String operators].
 
+[[functions-btrim]]
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/functions/string.adoc:22]:
```suggestion
[role=label--new-5.19]
[[functions-btrim]]
```

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc [hunk 2]
```diff
@@ -47,6 +47,21 @@ FINISH
...
+[source, cypher, role=noheader]
+----
+RETURN trim(BOTH 'x' FROM 'xxhelloxx'),
+       ltrim('xxhello', 'x'),
+       rtrim('helloxx', 'x'),
+       btrim('xxhelloxx', 'x')
+----
+| Extended the xref:functions/string.adoc#functions-trim[trim()], xref:functions/string.adoc#functions-ltrim[ltrim()]
+and xref:functions/string.adoc#functions-rtrim[rtrim()] functions to accept alternative trim character strings as well
+as introduced a new xref:functions/string.adoc#functions-btrim[btrim()] function.
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]:
```suggestion
| Introduced  xref:functions/string.adoc#functions-btrim[btrim()] function, which returns the given `STRING` with leading and trailing `trimCharacterString` characters removed.
Also extended the xref:functions/string.adoc#functions-trim[trim()], xref:functions/string.adoc#functions-ltrim[ltrim()], and xref:functions/string.adoc#functions-rtrim[rtrim()] functions to accept alternative trim character strings.
```

File: modules/ROOT/pages/functions/string.adoc [hunk 3]
```diff
@@ -90,13 +158,14 @@ RETURN left('hello', 3)
 [[functions-ltrim]]
 == ltrim()
 
-`ltrim()` returns the original `STRING` with leading whitespace removed.
+`ltrim()` returns the original `STRING` with leading `trimCharacterString` characters removed.
+If `trimCharacterString` is not specified then all leading whitespace will be removed.
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/functions/string.adoc]:
```suggestion
As of Neo4j 5.19, a `trimCharacterString` can be specified.
If this is not specified all leading whitespace will be removed.
```

File: modules/ROOT/pages/functions/string.adoc [hunk 4]
```diff
@@ -523,13 +598,14 @@ RETURN right('hello', 3)
 [[functions-rtrim]]
 == rtrim()
 
-`rtrim()` returns the given `STRING` with trailing whitespace removed.
+`rtrim()` returns the original `STRING` with trailing `trimCharacterString` characters removed.
+If `trimCharacterString` is not specified then all trailing whitespace will be removed.
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/functions/string.adoc]:
```suggestion
As of Neo4j 5.19, a `trimCharacterString` can be specified.
If this is not specified all trailing whitespace will be removed.
```

File: modules/ROOT/pages/functions/string.adoc [hunk 5]
```diff
@@ -977,13 +1060,14 @@ RETURN toUpper('hello')
 [[functions-trim]]
 == trim()
 
-`trim()` returns the given `STRING` with leading and trailing whitespace removed.
+`trim()` returns the given `STRING` with the leading and/or trailing `trimCharacterString` character removed.
+If `trimCharacterString` is not specified then all leading and/or trailing whitespace will be removed.
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/functions/string.adoc]:
```suggestion
As of Neo4j 5.19, a `trimCharacterString` can be specified.
If this is not specified all leading and/or trailing whitespace will be removed.
```

**Review by JPryce-Aklundh — COMMENTED** [review]
Thanks @gem-neo4j. I have a few suggestions, mainly about making versioning clearer, but in general it looks good :)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-936.surge.sh

Updated pages:
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-936.surge.sh/cypher-manual/5/deprecations-additions-removals-compatibility)
- [Functions](https://neo4j-docs-cypher-936.surge.sh/cypher-manual/5/functions)
- [String functions](https://neo4j-docs-cypher-936.surge.sh/cypher-manual/5/functions/string)

**Comment by gem-neo4j** [comment]
[GQL compliance - trim function](https://trello.com/c/oIrykkro/936-gql-compliance-trim-function)

**Comment by gem-neo4j** [comment]
Still waiting on runtime to review, so this missed cutoff ✂️

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-936.surge.sh

Updated pages:
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-936.surge.sh/cypher-manual/5/deprecations-additions-removals-compatibility)
- [Functions](https://neo4j-docs-cypher-936.surge.sh/cypher-manual/5/functions)
- [String functions](https://neo4j-docs-cypher-936.surge.sh/cypher-manual/5/functions/string)

**Review by JPryce-Aklundh — APPROVED** [review]

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

**Pull request merged**